### PR TITLE
Upgrade Nestjs, Replace Nrwl with Nx, Added additional linter rules,  closes #48, #43

### LINKED ISSUE #48: Upgrade Nrwl/nx to @nx with subpackages

### PR DESCRIPTION
**Pull request by NarHakobyan**

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
LMK if you need any reviews at some point :)

**Comment by NarHakobyan** [comment]
@KnorpelSenf Sure 😄 🙏, I will one the PR on Sunday.

**Comment by KnorpelSenf** [comment]
Did you have to commit the 150K lines of yarn to this repo?

File: README.md [hunk 1]
```diff
@@ -22,24 +22,24 @@ This project was generated using [Nx](https://nx.dev).
 
 ### Alpha Status Implications
 
--   Alpha testers are needed to guide the implementation requirements needed for production deployments
--   As a pre-release, the API will likely change -- be prepared for breaking changes as we develop the optimal solution
+- Alpha testers are needed to guide the implementation requirements needed for production deployments
+- As a pre-release, the API will likely change -- be prepared for breaking changes as we develop the optimal solution
```
Comment (KnorpelSenf) [inline README.md:26]: Is this still accurate after this is merged?

File: apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { createBotDecorator } from '../../utils';
+
+/**
+ * Handler for /help command.
+ *
+ * @see https://telegraf.js.org/#/?id=help
```
Comment (KnorpelSenf) [inline apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts:6]: What?

File: package.json [hunk 3]
```diff
@@ -1,71 +1,68 @@
...
+    "@nestjs/common": "^10.0.2",
+    "@nestjs/core": "^10.0.2",
+    "@nestjs/platform-express": "^10.0.2",
+    "axios": "^1.0.0",
     "firebase": "^9.9.3",
     "firebase-admin": "latest",
     "firebase-functions": "latest",
-    "grammy": "^1.11.0",
-    "lodash": "^4.17.21",
+    "grammy": "^1.19.0",
```
Comment (KnorpelSenf) [inline package.json:22]: Should grammY be made a peer dep so that users can install their own version of it?

**Review by KnorpelSenf — COMMENTED** [review]
This whole thing is huge and I understand very little of it. I don't use Nestjs and I don't plan on getting to know it, so I can hardly come up with any meaningful comments for this review. I have read all of the changes here and most of the code is straightforward enough that I can't spot any flaws. On the other hand, I really don't know how all of these pieces will play to together, but it really looks like you know what you're doing, so this LGTM.

File: apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts (hunk 2 above)
Comment (NarHakobyan) [reply apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts:6]: that was there :D

File: package.json (hunk 3 above)
Comment (NarHakobyan) [reply package.json:22]: yeah, but this is a worksapce package json, it's already done in grammy-nestjs/package.json

File: apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts (hunk 2 above)
Comment (KnorpelSenf) [reply apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts:6]: I see. It shouldn't. Could we remove it in this pull request? The telegraf documentation no longer exists.

Do you know how to find the equivalent links for grammY docs?

File: package.json (hunk 3 above)
Comment (KnorpelSenf) [reply package.json:22]: I see! Then nevermind

File: README.md (hunk 1 above)
Comment (NarHakobyan) [reply README.md:26]: I think no, but I will recheck

File: apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts (hunk 2 above)
Comment (KnorpelSenf) [reply apps/grammy-nestjs/src/decorators/listeners/admin.decorator.ts]: Thanks!

**Comment by NarHakobyan** [comment]
@KnorpelSenf if everything okay, then I will do some internal check related to package build/install and usage and after that I can merge it if you don't mind

**Review by KnorpelSenf — APPROVED** [review]
Sounds good, go ahead!

**Pull request merged**